lib/pc_patch.c

```diff
@@ -608,3 +608,54 @@ PCPOINT *pc_patch_pointn(const PCPATCH *patch, int n)
 	pcerror("%s: unsupported compression %d requested", __func__, patch->type);
 	return NULL;
 }
+
+/** set schema for patch */
+PCPATCH*
+pc_patch_set_pcid(PCPATCH *patch, const PCSCHEMA *new_schema, double defaultvalue)
+{
+	PCDIMENSION** new_dimensions = new_schema->dims;
+	PCDIMENSION* old_dimensions[new_schema->ndims];
+	PCPATCH_UNCOMPRESSED *paout = NULL;
+	PCPOINTLIST *opl, *npl;
+	PCPOINT *opt, *npt;
+	size_t i, j;
+	double val;
+
+	for ( j = 0; j < new_schema->ndims; j++ )
+	{
+		PCDIMENSION *ndim = new_dimensions[j];
+		old_dimensions[j] = pc_schema_get_dimension_by_name(
+				patch->schema, ndim->name);
+	}
+
+	// init point lists
+	opl = pc_pointlist_from_patch(patch);
+	npl = pc_pointlist_make(patch->npoints);
+
+	// build the new pointlist
+	for ( i = 0; i <patch->npoints; i++ )
+	{
+		opt = pc_pointlist_get_point(opl, i);
+		npt = pc_point_make(new_schema);
+
+		for ( j = 0; j < new_schema->ndims; j++ )
+		{
+			PCDIMENSION *ndim = new_dimensions[j];
+			PCDIMENSION *odim = old_dimensions[j];
+			if ( odim == NULL )
+				val = defaultvalue;
+			else
+				pc_point_get_double(opt, odim, &val);
+			pc_point_set_double(npt, ndim, val);
```

Comment:
Please show me how you'd write it exactly?

Like the following?

```py
    if ( pc_point_get_double(opt, odim, &val) == PC_FAILURE )
        val = defaultvalue;
    pc_point_set_double(npt, ndim, val);
```

Or as follows?

```py
    val = defaultval;
    pc_point_get_double(opt, odim, &val);
    pc_point_set_double(npt, ndim, val);
```

I like neither of these forms, because they're not explicit enougth to the code reader. The way it is right now makes it explicit that we use `defaultval` with the "old dimension" does not exist.

Do you have another form in mind?